app/subscriber/src/components/tool-bar/AddToReportMenu.tsx

```diff
@@ -0,0 +1,114 @@
+import { TooltipMenu } from 'components/tooltip-menu';
+import React from 'react';
+import { FaFileExport, FaPlay } from 'react-icons/fa6';
+import { useReports } from 'store/hooks';
+import { useProfileStore } from 'store/slices';
+import { IContentModel, IReportModel, Row } from 'tno-core';
+
+import * as styled from './styled';
+import { newReportInstance, toInstanceContent } from './utils';
+
+export interface IAddToReportMenuProps {
+  content: IContentModel[];
+}
+
+export const AddToReportMenu: React.FC<IAddToReportMenuProps> = ({ content }) => {
+  const [{ updateReport, findMyReports }] = useReports();
+  const [{ myReports }] = useProfileStore();
+  const [activeReport, setActiveReport] = React.useState<IReportModel | null>(null);
+  React.useEffect(() => {
+    if (!myReports.length) {
+      findMyReports().catch(() => {});
+    }
+    // Only do this on init.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  /** Adds the content to the active report. */
+  const addContentToReport = React.useCallback(
+    (section: string) => {
+      if (!activeReport) {
+        return;
+      }
+
+      const convertedContent = toInstanceContent(content, activeReport, section);
+
+      if (!!activeReport.instances.length) {
+        // get the latest instance content and append to it
+        const instContent = [
+          ...activeReport.instances[activeReport.instances.length - 1].content,
+          ...convertedContent,
+        ];
+
+        // find the latest instance and replace the content with old content appended with new
+        // this should handle create new instance if there is no instance
+        const instances = activeReport.instances.map((inst, index) => {
+          if (index === activeReport.instances.length - 1) {
+            return {
+              ...inst,
+              content: instContent,
+            };
+          } else {
+            return inst;
+          }
+        });
+        const report = {
+          ...activeReport,
+          instances: instances,
+        };
+        updateReport(report, true).catch(() => {});
+      }
+
+      if (!activeReport.instances.length) {
```

Comment:
Don't come up with a new way to do this.  The My Reports already has this logic all worked out.  When you edit a report it will automatically generate an instance if you don't have one already.

![image](https://github.com/bcgov/tno/assets/3180256/1a639412-15da-4827-bf98-8ac824bb9260)

Refer to ReportEdit.tsx

![image](https://github.com/bcgov/tno/assets/3180256/fd967c49-764b-4939-a809-9af4daaa0783)